tls/s2n_handshake_type.c

```diff
@@ -61,7 +61,7 @@ S2N_RESULT s2n_handshake_type_set_tls13_flag(struct s2n_connection *conn, s2n_tl
 
 bool s2n_handshake_type_check_tls13_flag(struct s2n_connection *conn, s2n_tls13_handshake_type_flag flag)
 {
-    return conn && s2n_connection_get_protocol_version(conn) >= S2N_TLS13
+    return s2n_connection_get_protocol_version(conn) >= S2N_TLS13
```

Comment:
Why was `conn &&` removed?

Does `s2n_connection_get_protocol_version` check for that, so it was redundant?
If that's the case, can we remove `s2n_connection != NULL &&` from the new harnesses:
- `tests/cbmc/proofs/s2n_is_hello_retry_message/s2n_is_hello_retry_message_harness.c` and
- `tests/cbmc/proofs/s2n_is_hello_retry_handshake/s2n_is_hello_retry_handshake_harness.c`?

Reply:
> Does s2n_connection_get_protocol_version check for that, so it was redundant?

Yes, so we can remove the check only for `s2n_is_hello_retry_handshake`.

Reply:
You're right. Yes, the other one doesn't call `s2n_connection_get_protocol_version`. Running low on caffeine 😅